### PR TITLE
Move WinRT extensions into vulkan_win32.h header

### PR DESCRIPTION
**Pull request by r-potter**
@lygstate This should move the WinRT extension into vulkan_win32.h

@oddhack @jjulianoatnv Could you signoff if you are happy with this change?

### CODE REVIEW

**Comment by oddhack** [comment]
That is fine in script terms though I haven't looked at the generated header files.

**Comment by r-potter** [comment]
Diff on the generated headers appears to be fine:

```
r.potter@SRUK-ws-1803:~/dev/vulkan/gen/include$ diff vulkan vulkan-win32
diff vulkan/vulkan_core.h vulkan-win32/vulkan_core.h
14003,14020d14002
< #define VK_NV_acquire_winrt_display 1
< #define VK_NV_ACQUIRE_WINRT_DISPLAY_SPEC_VERSION 1
< #define VK_NV_ACQUIRE_WINRT_DISPLAY_EXTENSION_NAME "VK_NV_acquire_winrt_display"
< typedef VkResult (VKAPI_PTR *PFN_vkAcquireWinrtDisplayNV)(VkPhysicalDevice physicalDevice, VkDisplayKHR display);
< typedef VkResult (VKAPI_PTR *PFN_vkGetWinrtDisplayNV)(VkPhysicalDevice physicalDevice, uint32_t deviceRelativeId, VkDisplayKHR* pDisplay);
<
< #ifndef VK_NO_PROTOTYPES
< VKAPI_ATTR VkResult VKAPI_CALL vkAcquireWinrtDisplayNV(
<     VkPhysicalDevice                            physicalDevice,
<     VkDisplayKHR                                display);
<
< VKAPI_ATTR VkResult VKAPI_CALL vkGetWinrtDisplayNV(
<     VkPhysicalDevice                            physicalDevice,
<     uint32_t                                    deviceRelativeId,
<     VkDisplayKHR*                               pDisplay);
< #endif
<
<
diff vulkan/vulkan_win32.h vulkan-win32/vulkan_win32.h
310a311,328
>
> #define VK_NV_acquire_winrt_display 1
> #define VK_NV_ACQUIRE_WINRT_DISPLAY_SPEC_VERSION 1
> #define VK_NV_ACQUIRE_WINRT_DISPLAY_EXTENSION_NAME "VK_NV_acquire_winrt_display"
> typedef VkResult (VKAPI_PTR *PFN_vkAcquireWinrtDisplayNV)(VkPhysicalDevice physicalDevice, VkDisplayKHR display);
> typedef VkResult (VKAPI_PTR *PFN_vkGetWinrtDisplayNV)(VkPhysicalDevice physicalDevice, uint32_t deviceRelativeId, VkDisplayKHR* pDisplay);
>
> #ifndef VK_NO_PROTOTYPES
> VKAPI_ATTR VkResult VKAPI_CALL vkAcquireWinrtDisplayNV(
>     VkPhysicalDevice                            physicalDevice,
>     VkDisplayKHR                                display);
>
> VKAPI_ATTR VkResult VKAPI_CALL vkGetWinrtDisplayNV(
>     VkPhysicalDevice                            physicalDevice,
>     uint32_t                                    deviceRelativeId,
>     VkDisplayKHR*                               pDisplay);
> #endif
>

```

File: scripts/genvk.py [hunk 1]
```diff
@@ -384,7 +384,7 @@ def makeGenOpts(args):
         [ 'vulkan_macos.h',       [ 'VK_MVK_macos_surface'        ], commonSuppressExtensions ],
         [ 'vulkan_vi.h',          [ 'VK_NN_vi_surface'            ], commonSuppressExtensions ],
         [ 'vulkan_wayland.h',     [ 'VK_KHR_wayland_surface'      ], commonSuppressExtensions ],
-        [ 'vulkan_win32.h',       [ 'VK_.*_win32(|_.*)', 'VK_EXT_full_screen_exclusive' ],
+        [ 'vulkan_win32.h',       [ 'VK_.*_win32(|_.*)', 'VK_.*_winrt(|_.*)', 'VK_EXT_full_screen_exclusive' ],
                                                                      commonSuppressExtensions +
```
Comment (lygstate) [inline scripts/genvk.py:388]: I am confused, is that possible check `platform="win32"` instead? That's more reliable.

**Review by dgkoch — APPROVED** [review]
Discussed with @pdaniell-nv and @jjulianoatnv and we are ok with this change.

Technically this extension does not have "win32" in the name because it is not a win32 extension. But we don't have a "winrt" header and we didn't want to create just for this extension, so we are ok with using win32.

**Comment by jjulianoatnv** [comment]
This extension creates interaction between Vulkan and a piece of the Windows OS surface named WinRT.  The WinRT API surface is not part of the win32 API surface. These are disjoint API surfaces both provided by the Windows OS.

So, technically, it is incorrect to call this a "win32" extension. That's why the extension name includes the substring "winrt" rather than the substring "win32".

However, it I do not wish to assert that a new header file should be added for only these two functions. For convenience, I'm OK with adding these functions into the win32 header file even though they are not "win32 functions".

**Comment by oddhack** [comment]
@chadversary could you sign off on this for the SI TSG? @dgkoch has approved for NVIDIA.

**Comment by oddhack** [comment]
> What is the historical reason that this header is named with the substring "win32" rather than with the substring "windows"?

That the WSI extensions all had 'win32' in their names. I don't know if that was to distinguish from some other set of Windows APIs or something else.

**Comment by lygstate** [comment]
> This extension creates interaction between Vulkan and a piece of the Windows OS surface named WinRT. The WinRT API surface is not part of the win32 API surface. These are disjoint API surfaces both provided by the Windows OS.
> 
> So, technically, it is incorrect to call this a "win32" extension. That's why the extension name includes the substring "winrt" rather than the substring "win32".
> 
> However, it I do not wish to assert that a new header file should be added for only these two functions. For convenience, I'm OK with adding these functions into the win32 header file even though they are not "win32 functions".
> 
> What is the historical reason that this header is named with the substring "win32" rather than with the substring "windows"?

`windows` are more confusing, as window-s, window is a general concept,
on all `MS Windows`  platform,  _WIN32 macro is always predefined, so indeed win32 is a more  concrete concept that specifically to `Microsoft Windows`

**Comment by oddhack** [comment]
The WG decided to go ahead with this without waiting for @chadversary since the SI TSG had effectively had input through NVIDIA already, anyway.

**Pull request merged**